modules/library/jdbc/src/main/java/org/geotools/jdbc/InsertionClassifier.java

```diff
@@ -0,0 +1,101 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2015, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.jdbc;
+
+import com.vividsolutions.jts.geom.Geometry;
+import org.geotools.factory.Hints;
+import org.opengis.feature.simple.SimpleFeature;
+import org.opengis.feature.simple.SimpleFeatureType;
+import org.opengis.feature.type.AttributeDescriptor;
+import org.opengis.feature.type.GeometryDescriptor;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * PreparedStatement inserts must be split in function of those criteria:
+ * - useExisting
+ * - type of the geometries
```

Comment:
Sadly, we do need to classify the geometries based on their types. At least for DB2. Please have a look at [DB2Util#prepareGeometryValue](https://github.com/geotools/geotools/blob/6b7fb3e34ccd12cb446c47ae472cbeb1b25dec55/modules/plugin/jdbc/jdbc-db2/src/main/java/org/geotools/data/db2/DB2Util.java#L157-157).